cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServlet.java

```diff
@@ -414,7 +414,7 @@ private void addOrRemoveBroker(HttpServletRequest request,
       return;
     }
 
-    CruiseControlResponse ccResponse = getAndMaybeReturnProgress(request, response, function.apply(parameters));
+    CruiseControlResponse ccResponse = getAndMaybeReturnProgress(request, response, function.apply(parameters), parameters);
```

Comment:
Can we refrain from passing parameters? Can't we extract the parameters for logging using the parameterMap from the request itself?

Reply:
The abstract method `AbstractCruiseControlResponse.discardIrrelevantAndCacheRelevant(CruiseControlParameters parameters)` where each subclass of `AbstractCruiseControlResponse` implement takes `CruiseControlParameters` type arguments. It's easier compared to changing the signature to add `isJson` and `isVerbose` for all the implementations. I think since it's the user task result that dominates memory usage, storing `CruiseControlParameters` in each `UserTaskInfo` to help with retrieval is okay here.